pennylane_qiskit/qiskit_device2.py

```diff
@@ -523,13 +522,22 @@ def execute_circuits(session):
 
     def _execute_runtime_service(self, circuits, session):
         """Execution using old runtime_service (can't use runtime sessions)"""
+
+        # The legacy ``backend.run()`` interface in Qiskit Runtime, which was used as the dedicated “direct hardware access” entry point, has been deprecated by Qiskit.
+        # The new SamplerV2 class now fulfills this role. Support for the backend.run() will be dropped on or around October 15, 2024.
+        # Please refer to the `migration guide <https://docs.quantum.ibm.com/api/migration-guides/qiskit-runtime>`_ for instructions on how to migrate any existing code.
+        # This corresponds to the "circuit-runner" and "qasm3-runner" programs if you are invoking the REST API directly.
+        # ToDo: deprecate this by or around October 15, 2024.
+
         # update kwargs in case Options has been modified since last execution
         self._update_kwargs()
 
         # in case a single circuit is passed
         if isinstance(circuits, QuantumScript):
             circuits = [circuits]
 
+        shots = circuits[0].shots.total_shots or self.shots.total_shots
```

Comment:
So, the underlying assumption is that all circuits (in the list) will have an equal number of `shots`?

Reply:
Suppose the user does something like `circuit(shots=[5, 10, 2])`, we raise a warning letting them know this is not supported due to the way qiskit takes in shots, and then run the circuit with the total amount of shots (17) in this case. If you wanted the exact functionality of circuit(shots=[5, 10, 2]), you just have to run the circuit three times manually e.g. circuit(shots=5), circuit(shots=10), circuit(shots=2). 

The reason is just because Qiskit doesn't support an array of shots like PL does, and the amount of work it would be to parse through an array of shots and send the job to qiskit 3 times and process the results would be challenging and not worth the time/effort.

We talked in more detail about how to track/deal with the number of shots in a different PR, but the solution is basically a compromise where we let the user run the circuit, but we also raise a warning explaining exactly what has happened and what to do instead.